spec/aria-reflection.html

```diff
@@ -0,0 +1,761 @@
+        </ul>
+        <p class="ednote">Editor's Note: Should we make an exception on the spelling of "placeholder" and capitalize the H anyway? Some developers
+          will expect this to be
+          <code>ariaPlaceHolder</code> despite the fact that it's not a hyphenated word.</p>
+      </section>
+    </section>
+
+    <section class="informative" id="idl_example_usage">
+      <h2>Example IDL Attribute Usage</h2>
+      <p>The primary purpose of ARIA IDL attribute reflection is to ease JavaScript-based manipulation of values. The following
+        examples demonstrate its usage.</p>
+      <aside class="example">
+
+        <!-- ReSpec needs these examples to be unindented. -->
+        <pre>&lt;div id="inaccessibleButton"&gt;
+    &lt;!-- Use semantic markup instead. This is just a retrofit example. --&gt;
+&lt;/div&gt;</pre>
+        <pre>// Get a reference to the element.
+let el = document.getElementById('inaccessibleButton'); 
+el.tabIndex = 0; // Make it focusable.
+
+// Set the role and label.
+el.role = "button";
+el.ariaLabel = "Edit";
+
+// Get the role and label.
+el.role; // Returns "button"
+el.ariaLabel; // Returns "Edit"
+
+// These are interchangeable with the more verbose setAttribute and getAttribute methods.
+el.setAttribute("role", "button");
+el.setAttribute("aria-label", "Edit");
+el.getAttribute("role"); // Returns "button"
+el.getAttribute("aria-label"); // Returns "Edit"
+
+// Changes via either interface are reflected by the other.
+el.setAttribute("aria-label", "Delete");
+el.ariaLabel; // Returns "Delete"
+el.ariaLabel = "Publish";
+el.getAttribute("aria-label"); // Returns "Publish"</pre>
+
+      </aside>
+    </section>
+
+    <section class="normative" id="shadowroot_interaction">
+      <h2>Interaction between
+        <code>ShadowRoot</code> and
+        <code>host</code>
+      </h2>
+      <p>If an
+        <code>Element</code> has an attached
+        <code>ShadowRoot</code>, and the author has access to the
+        <code>ShadowRoot</code>, authors may set a
+        <code>role</code> or ARIA attribute, such as
+        <a href="#dom-ariaattributes-ariaactivedescendant">
+          <code>ariaActivedescendant</code>
+        </a>, on
+        <em>either</em> the
+        <code>Element</code>,
+        <em>or</em> the
+        <code>ShadowRoot</code>, or both.
+      </p>
+      <p>
+        If a
+        <code>role</code> or ARIA attribute property is set on
+        <em>either</em> the
+        <code>Element</code>
+        <em>or</em> its associated
+        <code>ShadowRoot</code>, that property should be
+        <a href="https://www.w3.org/TR/core-aam-1.1/#mapping_general">mapped</a>
+        to the
+        <a href="https://www.w3.org/TR/core-aam-1.1/#dfn-accessible-object">accessible object</a>
+        <a href="https://www.w3.org/TR/core-aam-1.1/#include_elements">associated</a>
+        with the host
+        <code>Element</code>.
+      </p>
+      <p>
+        If a property is set on
```

Comment:
I think it'd be good to explicitly clarify what happens with the empty string vs null.

A reflected property can be set to null, which is the same as removing the attribute,
right? But setting it to the empty string overrides.

Reply:
Added a mention of non-null value here.